core/src/main/java/gyro/core/reference/FinderReferenceResolver.java

```diff
@@ -19,23 +19,29 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 import java.util.stream.Collectors;
 
 import gyro.core.GyroException;
 import gyro.core.Type;
+import gyro.core.auth.CredentialsSettings;
 import gyro.core.finder.Finder;
 import gyro.core.finder.FinderField;
 import gyro.core.finder.FinderSettings;
 import gyro.core.finder.FinderType;
 import gyro.core.resource.DiffableInternals;
 import gyro.core.resource.Resource;
 import gyro.core.scope.Scope;
+import gyro.lang.ast.value.ReferenceNode;
 
 @Type("external-query")
 public class FinderReferenceResolver extends ReferenceResolver {
 
     @Override
-    public Object resolve(Scope scope, List<Object> arguments) {
+    public Object resolve(ReferenceNode node, Scope scope, List<Object> arguments) {
```

Comment:
I think we should only pass `node` and remove `arguments` here. Provide a helper method for grabbing all arguments as well as individual arguments, just like `DirectiveProcessor`. Since the methods from `DirectiveProcessor` are just copies we should find a way to avoid duplication of code.